drivers/sensor/voltage_divider/voltage.c

```diff
@@ -100,6 +112,7 @@ static int __maybe_unused pm_action(const struct device *dev, enum pm_device_act
 		if (ret != 0) {
 			LOG_ERR("failed to set GPIO for PM resume");
 		}
+		wait_power_off_on_delay(config->off_on_delay_us);
```

Comment:
You could put this in an `else` branch or return before this if `ret != 0`.

If `gpio_pin_set_dt(&config->gpio_power, 1);` fails, there is no reason to sleep.